cmd/agola/cmd/runlist.go

```diff
@@ -82,13 +106,36 @@ func runList(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	runs := make([]*gwapitypes.RunResponse, len(runsResp))
+	runs := make([]*runDetails, len(runsResp))
 	for i, runResponse := range runsResp {
 		run, _, err := gwclient.GetRun(context.TODO(), runResponse.ID)
 		if err != nil {
 			return err
 		}
-		runs[i] = run
+
+		tasks := []*taskDetails{}
+		for _, task := range run.Tasks {
+			runTaskResponse, _, err := gwclient.GetRunTask(context.TODO(), run.ID, task.ID)
+			t := &taskDetails{
+				name:            task.Name,
+				level:           task.Level,
+				runTaskResponse: runTaskResponse,
+				retrieveError:   err,
+			}
+			tasks = append(tasks, t)
+		}
+
+		sort.Slice(tasks, func(i, j int) bool {
+			return tasks[i].name < tasks[j].name
+		})
+		sort.SliceStable(tasks, func(i, j int) bool {
+			return tasks[i].level < tasks[j].level
+		})
```

Comment:
I'm not sure this will work, SliceStable will use the stable sort algorithm that will keep the order of equal (type equality) elements and doesn't know that you previously ordered by name. You should do this in an single sort (that's also cleaner and faster):

```
		sort.Slice(tasks, func(i, j int) bool {
			if tasks[i].level != tasks[j].level {
				return tasks[i].level < tasks[j].level
			}
			return tasks[i].name < tasks[j].name
		})

```